File: TriangleType/MultiThreaded/EquilateralTriangle.cs
namespace MultiThreaded
{
    using Model;

    public class EquilateralTriangle : ITriangle
    {
        public bool DoMatch(TriangleSideLengths triangleSideLengths)
        {
            return triangleSideLengths.LengthOfSideA == triangleSideLengths.LengthOfSideB &&
                   triangleSideLengths.LengthOfSideB == triangleSideLengths.LengthOfSideC;
        }

        public string Description
        {
            get { return "All sides have the same length. That means that the triangle is equilateral."; }
        }
    }
}

File: TriangleType/ReactJS/ObjectOriented/EquilateralTriangle.js

function EquilateralTriangle(){
    this.Description = "All sides have the same length. That means that the triangle is equilateral.";
    this.DoMatch = function (triangleSideLengths){
        return triangleSideLengths.LengthOfSideA == triangleSideLengths.LengthOfSideB &&
               triangleSideLengths.LengthOfSideB == triangleSideLengths.LengthOfSideC;
    }
}